app/server-utils/url-to-iframe.ts

```diff
@@ -1,29 +1,33 @@
 /**
- * Replaces all of the URLs in text with iframes of the URLs
+ * Replaces all whitelisted URLs in the text with iframes of the URLs
  * @param text text in which URLs should be replaced
  */
 export const urlToIframe = (text: string): string => {
   const whitelistedHosts: Record<string, HostConfig> = {
     'aisafety.world': {sandboxValue: 'allow-scripts allow-same-origin'},
   }
 
-  // Regex is from: https://stackoverflow.com/a/26764609
-  const anchorTagRegex = new RegExp(/<a[\s]+([^>]+)>((?:.(?!<\/a>))*.)<\/a>/gi)
-  const matches = text.matchAll(anchorTagRegex)
+  // Regex to extract known links:
+  // \[               <-- start of markdown link, i.e. [name](url)
+  //   (              <-- capture the url
+  //     https?:\/\/  <-- check for both http and https
+  //     [\.\w]+?     <-- any string of alphanumerical characters and full stops
+  //     \/?[^\]]*    <-- optional query params
+  //   )
+  // \]
+  // \(
+  //     \1           <-- lookback to match whatever the first group caught
+  // \)
+  const matches = text.matchAll(/\[(https?:\/\/[.\w]+?\/?[^]]*?)\]\(\1\)/gi)
```

Comment:
do we want to replace only links in the `[url](url)` format on purpose, keeping `[text](url)` as text links?

Reply:
yes. The idea is that there are a whitelisted collection of urls that are to be transformed into iframes, but only if both the url and text are the same. So you could do `[here's a map](http://aisafety.world/)` in the main text, without it then being transformed into an iframe of the map